dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceService.java

```diff
@@ -1413,6 +1438,56 @@ else if (currCal.equals(sched)) {
         this.isLogRolloverScheduled = true;
     }
 
+    /**
+     * Schedule FFDC file age based deletion
+     */
+    private void scheduleFfdcFileDeletion(LogProviderConfigImpl config) {
+        long maxFfdcAge = config.getMaxFfdcAge();
+        Boolean startNow = config.getFfdcCleanupStartNow();
+
+        if (this.isFfdcCleanupScheduled) {
+            //Return if the ffdcMaxAge attribute doesn't change. Otherwise, cancel the schedule.
+            if (this.maxFfdcAge == maxFfdcAge) {
+                return;
+            } else {
+                ffdcCleanup_Timer.cancel();
+                ffdcCleanup_Timer.purge();
+                this.isFfdcCleanupScheduled = false;
+            }
+        }
+
+        if(maxFfdcAge < 0) {
+            return;
+        }
+
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.debug(tc, "Scheduling ffdc log cleanup...");
+            Tr.debug(tc, "maxFfdcAge=" + maxFfdcAge);
+        }
+
+        this.maxFfdcAge = maxFfdcAge;
+
+        //set calendar start time
+        Calendar sched = Calendar.getInstance();
+
+        if(!startNow) {
+            sched.set(Calendar.HOUR_OF_DAY, 0);
+            sched.set(Calendar.MINUTE, 0);
+            sched.add(Calendar.DATE, 1);                //The cleanup will run everyday at midnight.
+        }
+        else {
+            sched.add(Calendar.SECOND, 1);      //Used for test cases in order to trigger the cleanup event immediately.
```

Comment:
Do you think you will need to adjust the 1 second delay later in test?  If yes, I'd recommend to have a numeric delay value for the `ffdcCleanupStartNow` rather than a boolean.

Reply:
No I don't think so as the cleanup won't be triggered until the maxFfdcAge attribute is updated in the tests.